torch/csrc/autograd/engine.cpp

```diff
@@ -277,10 +300,18 @@ auto Engine::thread_main(
   // Why the test on graph_task->outstanding_tasks_?  See
   // Note [Reentrant backwards]
   while (!reentrant_thread || graph_task->outstanding_tasks_ > 0) {
-    NodeTask task = queue->pop();
+    std::unique_ptr<NodeTask> task{nullptr};
+    if (graph_task) {
+      task = queue->pop(std::chrono::microseconds(100));
```

Comment:
What's going on with the timeout here? There doesn't seem to be any explanation of it anywhere.